toxics/toxic.go

```diff
@@ -39,21 +40,30 @@ type StatefulToxic interface {
 	NewState() interface{}
 }
 
+// Bidirectional toxics operate on both TCP streams and allow state to be shared.
+// PipeRequest() will oparate on the upstream, while Pipe() will operate on the downstream.
+type BidirectionalToxic interface {
+	// Defines the packet flow through an upstream ToxicStub. Operates the same as Pipe().
+	PipeRequest(*ToxicStub)
```

Comment:
Yeah, probably a good idea. I was originally writing this as part of the HTTP toxic, so that's where the naming came from.